internal/activity.go

```diff
@@ -333,6 +322,21 @@ func WithLocalActivityTask(
 		tagWorkflowID, task.params.WorkflowInfo.WorkflowExecution.ID,
 		tagRunID, task.params.WorkflowInfo.WorkflowExecution.RunID,
 	)
+	startedTime := time.Now()
+	scheduleToCloseTimeout := task.params.ScheduleToCloseTimeout
+	startToCloseTimeout := task.params.StartToCloseTimeout
+
+	if startToCloseTimeout == 0 {
+		startToCloseTimeout = scheduleToCloseTimeout
+	}
+	if scheduleToCloseTimeout == 0 {
+		scheduleToCloseTimeout = startToCloseTimeout
+	}
+	deadline := calculateActivityDeadline(task.scheduledTime, startedTime, scheduleToCloseTimeout, startToCloseTimeout)
+	if task.attempt > 1 && !task.expireTime.IsZero() && task.expireTime.Before(deadline) {
```

Comment:
This is still an open issue if we should removed this/can remove this. This `expireTime` is calculated based on the `ScheduleToCloseTimeout` and the workflow time, but is only used on the second attempt. This has been basically unchanged for 4+ years, but seems wrong.

Reply:
While wrong, let's leave for now

Reply:
Yeah I agree, just wanted to call this out since  when we put Go  on Core this will be something we have  to deal with.